plasma_cash/root_chain/contracts/Libraries/RLP.sol

```diff
@@ -1,416 +1,381 @@
-     assembly {
-         b0 := byte(0, mload(start))
-     }
-     if (b0 < DATA_SHORT_START) {
-         memPtr = start;
-         len = 1;
-         return;
-     }
-     if (b0 < DATA_LONG_START) {
-         len = self._unsafe_length - 1;
-         memPtr = start + 1;
-     } else {
-         uint bLen;
-         assembly {
-             bLen := sub(b0, 0xB7) // DATA_LONG_OFFSET
-         }
-         len = self._unsafe_length - 1 - bLen;
-         memPtr = start + bLen + 1;
-     }
-     return;
- }
-
- // Assumes that enough memory has been allocated to store in target.
- function _copyToBytes(uint btsPtr, bytes memory tgt, uint btsLen) private constant {
-     // Exploiting the fact that 'tgt' was the last thing to be allocated,
-     // we can write entire words, and just overwrite any excess.
-     assembly {
-         {
-                 let i := 0 // Start at arr + 0x20
-                 let words := div(add(btsLen, 31), 32)
-                 let rOffset := btsPtr
-                 let wOffset := add(tgt, 0x20)
-             tag_loop:
-                 jumpi(end, eq(i, words))
-                 {
-                     let offset := mul(i, 0x20)
-                     mstore(add(wOffset, offset), mload(add(rOffset, offset)))
-                     i := add(i, 1)
-                 }
-                 jump(tag_loop)
-             end:
-                 mstore(add(tgt, add(0x20, mload(tgt))), 0)
-         }
-     }
- }
-
- // Check that an RLP item is valid.
-     function _validate(RLPItem memory self) private constant returns (bool ret) {
-         // Check that RLP is well-formed.
-         uint b0;
-         uint b1;
-         uint memPtr = self._unsafe_memPtr;
-         assembly {
-             b0 := byte(0, mload(memPtr))
-             b1 := byte(1, mload(memPtr))
-         }
-         if(b0 == DATA_SHORT_START + 1 && b1 < DATA_SHORT_START)
-             return false;
-         return true;
-     }
+library RLP {
+    uint constant DATA_SHORT_START = 0x80;
+    uint constant DATA_LONG_START = 0xB8;
+    uint constant LIST_SHORT_START = 0xC0;
+    uint constant LIST_LONG_START = 0xF8;
+
+    uint constant DATA_LONG_OFFSET = 0xB7;
+    uint constant LIST_LONG_OFFSET = 0xF7;
+
+
+    struct RLPItem {
+        uint _unsafe_memPtr;    // Pointer to the RLP-encoded bytes.
+        uint _unsafe_length;    // Number of bytes. This is the full length of the string.
+    }
+
+    struct Iterator {
+        RLPItem _unsafe_item;   // Item that's being iterated over.
+        uint _unsafe_nextPtr;   // Position of the next item in the list.
+    }
+
+    struct Tx {
+        uint prevBlock;
+        uint uid;
+        uint amount;
+        address newOwner;
+        address signer;
+    }
+
+    /* Public Functions */
+
+    function createTx(bytes memory txBytes, uint numItems)
+        internal
+        constant
+        returns (Tx memory)
+    {
+        RLPItem[] memory txList = toList(toRLPItem(txBytes), numItems);
```

Comment:
can we just hardcode numItems to 5? seems like other number is not gonna work anyway, then we don't need to pass in the param